SPHINXsys/src/shared/common/sph_data_conainers.h

```diff
@@ -0,0 +1,115 @@
+/**
+ * @file 	sph_data_conainers.h
+ * @brief 	Set up of basic data structure.
+ * @author	Luhui Han, Chi ZHang and Xiangyu Hu
+ */
+
+#ifndef SPH_DATA_CONTAINERS_H
+#define SPH_DATA_CONTAINERS_H
+
+#include "base_data_package.h"
+
+namespace SPH {
+	/**
+	 * @brief Preclaimed classes.
+	 */
+	class BaseMaterial;
```

Comment:
The reason why you like forward declaration is for compiling speed?

Reply:
No. It is due to cross referring between several classes such as body, material and particle. I am happy to learn how to decrease the usage of forward declare from you.